src/components/PeoplePage.tsx

```diff
@@ -1,29 +1,38 @@
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { usePeople } from '../store/PeopleContext';
 
 export const PeoplePage = () => {
+  const { isLoading, errorMsg, filteredPeople } = usePeople();
+
+  if (isLoading) {
+    return <Loader />;
+  }
+
   return (
     <>
       <h1 className="title">People Page</h1>
 
       <div className="block">
         <div className="columns is-desktop is-flex-direction-row-reverse">
-          <div className="column is-7-tablet is-narrow-desktop">
-            <PeopleFilters />
-          </div>
+          {!isLoading && (
+            <div className="column is-7-tablet is-narrow-desktop">
+              <PeopleFilters />
+            </div>
+          )}
 
           <div className="column">
             <div className="box table-container">
-              <Loader />
-
-              <p data-cy="peopleLoadingError">Something went wrong</p>
-
-              <p data-cy="noPeopleMessage">There are no people on the server</p>
-
-              <p>There are no people matching the current search criteria</p>
-
-              <PeopleTable />
+              {!isLoading && errorMsg && (
+                <p data-cy="peopleLoadingError" className="has-text-danger">
+                  {errorMsg}
+                </p>
+              )}
+              {!isLoading && !errorMsg && filteredPeople.length === 0 && (
+                <p data-cy="noPeopleMessage">{errorMsg}</p>
+              )}
```

Comment:
The condition for displaying the 'no people' message uses `errorMsg`, which might not be appropriate if there is no error but simply no people to display. Consider using a different message or condition to handle this case.